integreat_compass/cms/views/offers/offer_form_view.py

```diff
@@ -0,0 +1,178 @@
+                "offer_form": offer_form,
+                "offer_version_form": offer_version_form,
+                "contact_form": contact_form,
+                "location_form": location_form,
+                "organization_form": organization_form,
+            },
+        )
+
+    # pylint: disable=too-many-locals
+    @transaction.atomic
+    def post(self, request, *args, **kwargs):
+        r"""
+        Submit :class:`~integreat_compass.cms.forms.offers.contact_form.ContactForm`,
+               :class:`~integreat_compass.cms.forms.offers.location_form.LocationForm`,
+               :class:`~integreat_compass.cms.forms.offers.offer_form.OfferForm`,
+               :class:`~integreat_compass.cms.forms.offers.offer_version_form.OfferVersionForm` and
+               :class:`~integreat_compass.cms.forms.offers.organization_form.OrganizationForm`
+
+        :param request: The current request
+        :type request: ~django.http.HttpRequest
+
+        :param \*args: The supplied arguments
+        :type \*args: list
+
+        :param \**kwargs: The supplied keyword arguments
+        :type \**kwargs: dict
+
+        :raises ~django.core.exceptions.PermissionDenied: If user does not have the permission to edit the specific page
+
+        :return: The rendered template response
+        :rtype: ~django.template.response.TemplateResponse
+        """
+        offer_instance = Offer.objects.filter(id=kwargs.get("offer_id")).first()
+        if not request.user.has_perm("cms.edit_offer", offer_instance):
+            raise PermissionDenied(
+                f"{request.user!r} does not have the permission to edit {offer_instance!r}"
+            )
+
+        user = User.objects.get(pk=request.user.id)
+
+        offer_form = OfferForm(
+            data=request.POST, additional_instance_attributes={"user": user}
+        )
+        offer_version_form = OfferVersionForm(
+            data=request.POST, additional_instance_attributes={"user": user}
+        )
+        contact_form = ContactForm(
+            data=request.POST, additional_instance_attributes={"user": user}
+        )
+        location_form = LocationForm(
+            data=request.POST, additional_instance_attributes={"user": user}
+        )
+        organization_form = OrganizationForm(
+            data=request.POST, additional_instance_attributes={"user": user}
+        )
+
+        forms = (
+            offer_form,
+            offer_version_form,
+            contact_form,
+            location_form,
+            organization_form,
+        )
+        if not all(form.is_valid() for form in forms):
+            for form in forms:
+                form.add_error_messages(request)
+        else:
+            print(contact_form.is_valid())
+            contact_instance = contact_form.save()
+            location_instance = location_form.save()
+            organization_instance = organization_form.save()
+
+            offer_instance = offer_form.save(commit=False)
```

Comment:
I think this is all done, so the foreign key constraints are not violated? Isn't there a better way to achieve this? Some django-function?

Reply:
To my knowledge, the `commit=False` flag already is the "django-way" of handling this use case, see [docs](https://docs.djangoproject.com/en/4.2/topics/forms/modelforms/#the-save-method).

Reply:
I think on its own this way is already rather concise, it's just that we have to do it for multiple forms which results in annoyingly many lines.

Reply:
Ah, but is there really a reason for `commit=False` in the first place?
Why not just do
```
            offer_form.instance.offer_contact = contact_form.save()
            offer_form.instance.location = location_form.save()
            offer_form.instance.organization = organization_form.save()
            offer_form.save()
```

instead of 
```
            offer_instance = offer_form.save(commit=False)
            offer_instance.offer_contact = contact_form.save()
            offer_instance.location = location_form.save()
            offer_instance.organization = organization_form.save()
            offer_instance.save()
            offer_form.save_m2m()
```
?

In think in my comment above I overlooked the fact that only many-to-many relationships require the object to be saved before the relationship can be assigned, for simple foreign keys you can just assign the fields of an unsaved model instance.

Reply:
Yep, that works as well! Pushed the change.